apps/redi-connect/src/pages/app/profile/Profile.tsx

```diff
@@ -89,7 +90,12 @@ function Profile() {
 
   const shouldHidePrivateContactInfo = currentUserIsMentee && !isAcceptedMatch
 
-  const viewMode = determineViewMode(profile)
+  const activeMentorshipMatch = 
+    myProfile.mentorshipMatches.find(match =>
+      match.status === MentorshipMatchStatus.Accepted
+    )
```

Comment:
Since we are repeating this (and similar derivations from profiles) in many places, I was wondering if we can come up with a context that these are calculated once, and shared all over the codebase. Just an idea for later 💡

Reply:
Nice idea and I have a few more - let's talk in our next Thursday morning call?